src/include/distributed/listutils.h

```diff
@@ -80,6 +80,104 @@ typedef struct ListCellAndListWrapper
 		 (((var) = lfirst_oid(var ## CellDoNotUse)) || true); \
 		 var ## CellDoNotUse = lnext_compat(l, var ## CellDoNotUse))
 
+/*
+ * foreach_ptr_modify -
+ *	  a convenience macro which loops through a pointer List and can modify list
```

Comment:
Perhaps instead of modify we call this "append"

We're doing a pretty sketchy thing here, which will behave differently on PG12 and PG13. For instance, if I insert at the start of the list in the loop, PG13 will see the current element twice, while PG12 will not.

It seems behaviour would be the same only for the append operation.

(In general, we should discourage this and use dlist, but at the moment it seems like the most practical solution to get to a release)

Reply:
Changed the name to `foreach_ptr_append`

I agree with dlist approach being the best solution here.